File: demo/python/query_product.py
import time

from trader import Trader


class TraderQueryProduct(Trader):
    def __init__(self, addr):
        Trader.__init__(self, addr=addr)


if __name__ == '__main__':
    addr = "127.0.0.1:8001"

    trader = TraderQueryProduct(addr)

    trader.query_product(
        qry_id="1",
        user_id="weidaizi",
        market="ctp",
        exchange="SHFE",
        type="future",
        symbol="rb",
        contract="",
    )
    time.sleep(1)
    trader.query_product(
        qry_id="2",
        user_id="weidaizi",
        market="ctp",
        exchange="SHFE",
        type="future",
        symbol="rb",
        contract="1901",
    )
    time.sleep(3)
    trader.query_product(
        qry_id="3",
        user_id="weidaizi",
        market="ctp",
        exchange="",
        type="",
        symbol="",
        contract=""
    )

    trader.message_loop()

File: demo/go/query_product/query_product.go
package main

import (
	"log"
	"time"

	DemoTrade "github.com/MuggleWei/babel-trader/demo/go/demo_trade"
	common "github.com/MuggleWei/babel-trader/src/babeltrader-common-go"
)

func main() {
	log.SetFlags(log.LstdFlags | log.LUTC | log.Lmicroseconds | log.Lshortfile)

	// ctp
	addr := "127.0.0.1:8001"
	qry_symbol := common.MessageQuery{
		QueryId:  "1",
		UserId:   "weidaizi",
		Market:   common.Market_CTP,
		Exchange: common.Exchange_SHFE,
		Symbol:   "rb",
	}

	qry_contract := common.MessageQuery{
		QueryId:  "2",
		UserId:   "weidaizi",
		Market:   common.Market_CTP,
		Exchange: common.Exchange_SHFE,
		Symbol:   "rb",
		Contract: "1905",
	}

	service := DemoTrade.NewDemoTradeService()
	service.Run(addr, func() {
		service.ReqQueryProduct(&qry_symbol)
		time.Sleep(time.Second * 3)
		service.ReqQueryProduct(&qry_contract)
	})
}
